snapcraft/_schema.py

```diff
@@ -42,7 +42,7 @@ def __init__(self, snapcraft_yaml=None):
 
     @property
     def part_schema(self):
-        sub = self._schema['properties']['parts']['patternProperties']
+        sub = self.schema['properties']['parts']['patternProperties']
```

Comment:
can you explain why is this change in here?
I find it weird that we will have an attribute part_schema and another schema.